cimgen/languages/modernpython/utils/base.py

```diff
@@ -133,38 +134,221 @@ def cgmes_attributes_in_profile(self, profile: BaseProfile | None) -> dict[str,
             for f in fields(parent):
                 shortname = f.name
                 qualname = f"{parent.apparent_name()}.{shortname}"  # type: ignore
+                infos = dict()
+
                 if f not in self.cgmes_attribute_names_in_profile(profile) or shortname in seen_attrs:
-                    # Wrong profile or already found from a parent.
                     continue
-                else:
-                    # Namespace finding
-                    # "class namespace" means the first namespace defined in the inheritance tree.
-                    # This can go up to Base, which will give the default cim NS.
-                    if (extra := getattr(f.default, "json_schema_extra", None)) is None:
-                        # The attribute does not have extra metadata. It might be a custom atttribute
-                        # without it, or a base type (int...).
-                        # Use the class namespace.
-                        namespace = self.namespace
-                    elif (attr_ns := extra.get("namespace", None)) is None:
-                        # The attribute has some extras, but not namespace.
-                        # Use the class namespace.
-                        namespace = self.namespace
-                    else:
+
+                # Namespace finding
+                # "class namespace" means the first namespace defined in the inheritance tree.
+                # This can go up to Base, which will give the default cim NS.
+                infos["namespace"] = self.namespace
+                extra = getattr(f.default, "json_schema_extra", None)
+                if extra is not None and extra.get("is_used"):
+                    # adding the extras, used for xml generation
+                    extra_info = {
+                        "attr_name": qualname,
+                        "is_class_attribute": extra.get("is_class_attribute"),
+                        "is_enum_attribute": extra.get("is_enum_attribute"),
+                        "is_list_attribute": extra.get("is_list_attribute"),
+                        "is_primitive_attribute": extra.get("is_primitive_attribute"),
+                        "is_datatype_attribute": extra.get("is_datatype_attribute"),
+                        "attribute_class": extra.get("attribute_class"),
+                    }
+                    if (extra.get("attribute_namespace", None)) is not None:
```

Comment:
You are correct `wrapped_class_comment` shoud be used in enum_template. In the same way `wrapped_comment` should be implemented for enum instances as it is not the case currently in `cimgen.py`

Concerning your test in `test_load_all.py`, it tries to instantiate all classes in resource folder. However, classes generated with `primitive_template.mustache`, `dataclass_template.mustache` and `enum_template.mustache` cannot be instantiated. 
One of the goals of the previous PR was to avoid creating classes that are then never instantiated.
- Classes such as Integer or Float are therefore instances of the class Primitive.
- Datatype classes such as CurrentFlow are instances of the class CIMDatatype. Theses classes provide information on a value (unit, multiplier, ...).
- Enum classes such as UnitSymbol cannot be instantiated either. 

To test these classes the best way is to test their attributes directly. 

I will have a look at the problem with pyright type test

Reply:
I fixed the pyright type test error by setting the `attribute_class` to the name of the class instead of the object itself.
It will make it less straightforward to retrieve the `unit`, `multipler` or `type` of the `attribute_class` but fixes the issue and simplifies the `class_template.mustache`